sklearn/utils/tests/test_multiclass.py

```diff
@@ -204,6 +206,16 @@ def test_unique_labels():
     assert_array_equal(unique_labels(np.ones((4, 5)), np.ones((5, 5))), np.arange(5))
 
 
+def test_unique_labels_and_samples():
+    """Raises a warning when the number of unique classes is
+    larger than 50% of `n_samples`"""
+
+    y = np.array([0, 1, 1, 2, 3, 4, 5])
+
+    with warnings.catch_warnings():
```

Comment:
I would rewrite this using `pytest.warns` and check that the message is what you expect. Documentation on how to do that: https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns